scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/handler/MainLambdaHandler.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.schedulingmanager.deployment.handler;
+
+import com.amazonaws.blox.schedulingmanager.deployment.DeploymentWorkflowApplication;
+import com.amazonaws.blox.schedulingmanager.deployment.exception.HandlerNotFoundException;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MainLambdaHandler extends DeploymentWorkflowApplication
```

Comment:
Why does this extend the application configuration class?

Reply:
To instantiate applicationcontext. Previously when I had multiple lambda functions I didn't want that code everywhere, now that we have one main lambda function, I can put 

  public static final ApplicationContext applicationContext =
      new AnnotationConfigApplicationContext(DeploymentWorkflowApplication.class);
here

Reply:
Makes sense. Might it then also make sense to move the `Map<String, Handler> handlers` to a simple field on this class, so that we don't have to deal with injection for that list?

Reply:
What I can do is just initialize the context in this class and do the routing in a different class that is ready to be injected.